### PR TITLE
718 requisition query tests

### PR DESCRIPTION
**Pull request by andreievg**
closes https://github.com/openmsupply/remote-server/issues/718

I'll make an issue to refactor test data for requisitions, I think at the end (downstream prs) it was getting a little bit messy.

otherwise pretty standard test with extras
* loader tests
* making sure sort and filter field mapping is tested

### CODE REVIEW

File: repository/src/mock/common.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        .upsert_one(&invoice.invoice)
+        .unwrap();
+    for line in invoice.lines.iter() {
+        StockLineRowRepository::new(&connection)
+            .upsert_one(&line.stock_line)
+            .unwrap();
+        InvoiceLineRowRepository::new(&connection)
+            .upsert_one(&line.line)
+            .unwrap();
+    }
```
Comment (clemens-msupply) [inline repository/src/mock/common.rs:56]: why did you move it into common.rs?

File: repository/src/mock/test_requisition_queries.rs [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+                requisition_id: requisition_id.clone(),
+                item_id: mock_item_b().id,
+                requested_quantity: 15,
+                calculated_quantity: 3,
+                supply_quantity: 0,
+                stock_on_hand: 1,
+                average_monthly_consumption: 10,
+            },
+        ],
+    }
```
Comment (clemens-msupply) [inline repository/src/mock/test_requisition_queries.rs:96]: Looks nice to have everything defined in one struct

File: repository/src/mock/test_requisition_queries.rs [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+            status: RequisitionRowStatus::Draft,
+            created_datetime: NaiveDate::from_ymd(2021, 01, 01).and_hms(0, 0, 0),
+            sent_datetime: Some(NaiveDate::from_ymd(2021, 01, 02).and_hms(0, 0, 0)),
+            finalised_datetime: Some(NaiveDate::from_ymd(2021, 01, 03).and_hms(0, 0, 0)),
+            colour: Some("colour".to_owned()),
+            comment: Some("comment".to_owned()),
+            their_reference: Some("their_reference".to_owned()),
+            max_months_of_stock: 1.0,
+            threshold_months_of_stock: 0.9,
+            linked_requisition_id: None,
```
Comment (clemens-msupply) [inline repository/src/mock/test_requisition_queries.rs:72]: what is a linked requisition? is it the request for a response?

**Review by clemens-msupply — APPROVED** [review]
LG

btw. had a look and it should be very easy to move the graphql tests from the server crate into the graphql crate.

File: repository/src/mock/common.rs (hunk 1 above)
Comment (andreievg) [reply repository/src/mock/common.rs:56]: because the type is being used in mock data outside of full_invoice

File: repository/src/mock/test_requisition_queries.rs (hunk 2 above)
Comment (andreievg) [reply repository/src/mock/test_requisition_queries.rs:96]: thanks, i agree

File: repository/src/mock/test_requisition_queries.rs (hunk 3 above)
Comment (andreievg) [reply repository/src/mock/test_requisition_queries.rs:72]: yeah vise versa, for request it's the response and for response it's the request

**Comment by andreievg** [comment]
> btw. had a look and it should be very easy to move the graphql tests from the server crate into the graphql crate.

Nice, i think we can do that when we re-arrange graphl crates

**Comment by andreievg** [comment]
merged via: https://github.com/openmsupply/remote-server/pull/774, will keep an eye on further comments though
